File: java/src/main/java/testingil/webinar/tddapi/sfinal/NavigationController.java
package testingil.webinar.tddapi.sfinal;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/navigator")
public class NavigationController {

	
	
	@Autowired
	DistanceProvider distProvider;
	@Autowired
	Navigator nav;
	
	@PostMapping("/destination")
	public void setDestination(@RequestBody Location location) {
		nav.setDestination(location);
	}
	
	@PostMapping("/startpoint")
	public void setStartPoint(@RequestBody Location location) {
		nav.setStartingPoint(location);
	}
	
	@GetMapping(value = "/distance", produces = MediaType.APPLICATION_JSON_VALUE)
	public Distance getDistance() {
		return nav.getDistanceFromDestination();
	}
}

File: Python/s17_fill_controller/navigation_controller.py
from flask import Flask, request

from s17_fill_controller.distance_adapter import DistanceAdapter
from s17_fill_controller.location import Location
from s17_fill_controller.navigator import Navigator

app = Flask(__name__)
nav: Navigator


@app.route('/nav/destination', methods=['POST'])
def set_destination():
    content = request.json
    dest = Location(content['city'])
    nav.set_destination(dest)


@app.route('/nav/startpoint', methods=['POST'])
def set_start_point():
    content = request.json
    start = Location(content["city"])
    distProvider = DistanceAdapter()
    nav = Navigator(start, distProvider)
    return ('', 200)

@app.route('/nav/distance', methods=['GET'])
def get_distance(self):
    return nav.get_distance_from_destination()
